### PR TITLE
Xor extension for Snarky

### PR DESCRIPTION
**Pull request by querolita**
This PR adds extension of Xor16 gate types in Snarky. A preliminary version of this PR created a Generic gate right after the Xor16 gate, as it was supposed to be a Xor16-only gadget. But since we have moved to a "minimal" design, we will proceed with more complex gadgets later and this is just an exposition of the gate types from Kimchi to Snarky.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

* Closes #0000

### CODE REVIEW

**Comment by querolita** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by querolita** [comment]
!ci-build-me

File: src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml [hunk 1]
```diff
@@ -1156,6 +1214,7 @@ end = struct
              ; None
              ; None
              ; None
+             ; None
```
Comment (jspada) [inline src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml:1217]: Nice catch!!!

**Comment by querolita** [comment]
!ci-build-me

**Comment by querolita** [comment]
!ci-build-me

**Comment by querolita** [comment]
!ci-build-me

**Comment by querolita** [comment]
!ci-build-me

**Pull request merged**